reana-ui/src/pages/workflowDetails/components/LauncherLabel.js

```diff
@@ -0,0 +1,46 @@
+/*
+  This file is part of REANA.
+  Copyright (C) 2022 CERN.
+
+  REANA is free software; you can redistribute it and/or modify it
+  under the terms of the MIT License; see LICENSE file for more details.
+*/
+
+import PropTypes from "prop-types";
+import { Label } from "semantic-ui-react";
+
+export default function LauncherLabel({ url }) {
+  /**
+   * Return appropriate Label component props according to the given launcher URL.
+   * @param {String} url Submission workflow launcher URL.
+   * @returns {Object} Appropriate component props.
+   */
+  function getProps(url) {
+    const externalSrcProps = {
+      as: "a",
+      href: url,
+      target: "_blank",
+      rel: "noopener noreferrer",
+    };
+
+    if (!url) {
+      return { content: "reana-client", icon: "terminal" };
+    } else if (url.startsWith("https://zenodo.org/")) {
+      return { content: "Zenodo", color: "blue", ...externalSrcProps };
+    } else if (url.startsWith("https://github.com/")) {
```

Comment:
BTW one side musing about the "URL" label. Considering that it is short and people don't know where it points upn first look, and considering that elsewhere on the page we display "labels with details" regarding environment for example:

![b5](https://user-images.githubusercontent.com/517546/157294894-b943c88f-4082-424e-83fc-07a16871d3f3.png)

I wonder whether it would be an option to display the first N letters of the URL?  And name of the GitHub repository perhaps?

Just wondering out loud, because "wide label" would on one hand bring more information, but on the other hand it would make the page labels look too "wide" everywhere with those job labels below. I actually like how the current labels look; it's aesthetically very nicely complementing those "wide" labels below. So I'm just wondering how we could present the URL they point to more easily... Perhaps by a hover effect as when we pass the cursor over "Finished 6 minutes ago" and we see exact "MM DD YYYY HH:MM" popping up?

Reply:
> Minor: perhaps we should use `includes()` here? Since for example `https://raw.githubusercontent.com/reanahub/reana-demo-root6-roofit/master/reana.yaml` would not be displayed with `github` label

I think that the `URL` label for that particular URL is correct, as it's acting as a _kind of a CDN_ rather than a repo. If you check the `fetcher` logic, the `WorkflowFetcherGit` is only used for URLs that end with `.git` and URLs whose Network location part (aka domain) is `github.com`:

https://github.com/reanahub/reana-server/blob/b90d2513280c2834b86d1d6b7c0f67d077a92b16/reana_server/fetcher.py#L237-L240

I decided to go with `startsWith("https://...")` instead of `includes("github")` because `includes("github")` might trigger false positives such as e.g. `https://mycdn.org/repos/github/roofit/reana-cwl.yaml` that would qualify as "GitHub" instead of URL.

Said that, perhaps it's cleaner to [parse the URL](https://developer.mozilla.org/en-US/docs/Web/API/URL) to better identify the domain and so on. However, this raises again my concern mentioned during one of the stand-ups: does it make sense to duplicate the logic both in server and UI to identify the origin of a given URL?

Reply:
> Also a trailing slash makes a difference. Currently, if you specify `https://github.com/reanahub/reana-demo-root6-roofit/` you get a GitHub label, and if you specify `https://github.com/reanahub/reana-demo-root6-roofit` you get a URL label. It would be good to get the GitHub label in both cases.

Are you sure? For me both URLs are displaying the "GitHub" label. If you check the code we're only checking `url.startsWith("https://github.com/")`.

Reply:
> I wonder whether it would be an option to display the first N letters of the URL? And name of the GitHub repository perhaps?

Showing the actual URL in a popup should be pretty simple and would avoid displaying a full URL (or partial) on a label, that in my opinion would look rather ugly. Displaying the URL in a popup would also cover displaying the name of the GitHub repo somehow.